stigma/src/main/java/io/nixer/nixerplugin/stigma/decision/StigmaDecision.java

```diff
@@ -0,0 +1,76 @@
+package io.nixer.nixerplugin.stigma.decision;
+
+import java.util.StringJoiner;
+import java.util.function.Consumer;
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.Immutable;
+
+import com.google.common.base.Preconditions;
+import io.nixer.nixerplugin.stigma.domain.RawStigmaToken;
+import org.springframework.util.Assert;
+
+/**
+ * Represents decision whether a Stigma is required to be refreshed or not.
+ * If the refresh is required, by means of the {@link #event}, the carried {@link #tokenForRefresh} is going to be used.
+ * <br>
+ * The {@link #tokenForRefresh} is considered <b>always valid and ready to be used for refreshing</b>, despite the decision.
+ * It might be a brand new token or a valid, already being used, one.
+ * <br>
+ *
+ * Created on 2019-04-29.
+ *
+ * @author gcwiak
+ */
+@Immutable
+public class StigmaDecision {
+
+    @Nonnull
+    private final RawStigmaToken tokenForRefresh;
+
+    @Nonnull
+    private final StigmaEvent event;
+
+    public StigmaDecision(@Nonnull final RawStigmaToken tokenForRefresh, @Nonnull final StigmaEvent event) {
+        this.tokenForRefresh = Preconditions.checkNotNull(tokenForRefresh, "tokenForRefresh");
+        this.event = Preconditions.checkNotNull(event, "event");
+    }
+
+    /**
+     * Applies decision by executing the given action if token refresh is required or ignoring it otherwise.
+     *
+     * @param refreshAction to be executed, never null.
+     */
+    public void apply(@Nonnull final Consumer<RawStigmaToken> refreshAction) {
```

Comment:
Only one consumer implementation is injected into this apply() method. And there is little chance that a different one will be injected in future. 
In such scenario I would rather use pre-java8 code structure.
Also _apply()_ name can be changed to _refresh()_ as this is the purpose of this method.

Reply:
Not a big fan of methods accepting functions when only one type of function is injected. (Unless the function injected is used for numerous injections in different places)

Reply:
My intention here was not reusability but rather simpler API by enclosing all details inside the decision, e.g. not exposing the token as it might be not necessary at all.
Though I see your point, also after some more thinking I decided to bring back the original approach as maybe a bit more clumsy but more straightforward than another level of indirection.